src/component/main-content/main-content.js

```diff
@@ -15,10 +15,37 @@ export default class MainContent extends React.Component {
   componentWillMount() {
     this.setState({
       filter: null,
-      activeRequest: null
+      activeRequest: null,
+      contextMenuRequest: null
     });
   }
 
+  setActiveRequest(request) {
+    this.setState({activeRequest: request});
+  }
+
+  isActiveRequest(request) {
+    if (!this.state.activeRequest || !request) {
+      return false;
+    }
+    return this.state.activeRequest.request.id === request.request.id;
+  }
+
+  setContextMenuRequest(request) {
+    if (!request || request.request.id === this.state.contextMenuRequest) {
```

Comment:
If I'm setting the `contextMenuRequest` to the current `contextMenuRequest`, then clear the current the `contextMenuRequest`? This isn't super understandable